### ISSUE TITLE
dev: bump starknet-rs to 0.6

### ISSUE THREAD

**Issue by Eikix** [kkrt-labs/kakarot-rpc#528]
**Describe the enhancement request**

Starknet has just released 0.6 version.
It bumps the JSON RPC spec to version 0.4.

```[tasklist]
### Tasks
- [ ] Check the PR that bumps the spec to 0.4 and updates all type: https://github.com/xJonathanLEI/starknet-rs/commit/e9bdd7efeae9c76ef62a130ccef3f4355a864797
- [ ] Bump the starknet package in RPC repos from 0.4 to 0.6
- [ ] Fix types
```

**Comment by d-roak** [comment]
assign me

**Comment by d-roak** [comment]
blocked by: https://github.com/dojoengine/dojo/pull/867

**Comment by Eikix** [comment]
thanks @d-roak ! do you know if madara also has transitioned to starknet-rs 0.6? if it plans to?

> blocked by: [dojoengine/dojo#867](https://github.com/dojoengine/dojo/pull/867)

**Comment by d-roak** [comment]
We already did, if I'm not mistaken

**Comment by kariy** [comment]
merged `0.4.0` rpc in https://github.com/dojoengine/dojo/pull/867

**Comment by Eikix** [comment]
> merged `0.4.0` rpc in [dojoengine/dojo#867](https://github.com/dojoengine/dojo/pull/867)

opayaaa
